### DISCUSSION TITLE
Foreman & MRSK incompatibility – best way to approach?

### DISCUSSION THREAD

**Discussion by Morozzzko** [basecamp/kamal#420] [Q&A]
Hey y'all!

So I'm running a pretty barebones Rails 7.0.6 project, generated with `rails new --skip-action-mailer --skip-action-mailbox --skip-action-text --skip-active-storage --no-skip-hotwire --css=tailwind --database=postgresql --skip-test --main`

It also generated a `Procfile.dev` for me. [Rails docs](https://github.com/rails/rails/blob/3b5a4a56fb1dc471fa6dd9450d8ed50d72d45732/guides/source/working_with_javascript_in_rails.md?plain=1#L85-L86) say that I should be using `bin/dev` to run dev server in this case.

However, when I tried running dev server, I got a pretty weird error:

```

ActiveRecord::ConnectionNotEstablished
could not connect to server: Operation timed out
Is the server running on host "10.110.0.3" and accepting
TCP/IP connections on port 25061? 
```

Yet, `bin/rails s` works fine

Spent some time debugging and realized that it's trying to connect to my `DATABASE_URI` from my `.env` – Foreman seems to load it automatically. 

Now I'm confused – I'm running vanilla Rails and pretty simple config of MRSK. Yet, those offer incompatible solutions

What's the best way to bring back `bin/dev`, then? 

Basically, all the context:

* Rails 7.0.6 with Tailwindcss bundling
* MRSK 0.15.1
* My only `.env` file is for MRSK – I don't use dotenv

**Comment by igor-alexandrov** [comment]
Do you need `.env` for local build and deployment with MRSK?

**Comment by morgoth** [comment]
You could create a local .env-dev or similar and use that one for the foreman.
To do that you need to modify `bin/dev` and change the foreman line code to `exec foreman start -e .env-dev -f Procfile.dev "$@"`

**Reply by Morozzzko** [reply]
It appears that I'll have to manually intervene to make it work with `dotenv` and other `.env`-handling tooling. Feels a bit odd as far as the defaults are concerned

**Comment by Flixt** [comment]
Here to cross reference https://github.com/rails/rails/issues/51829. There really seem to be some problems related to Kamal/Foreman/DotEnv confusions.

**Comment by morgoth** [comment]
This will be handled in https://github.com/basecamp/kamal/pull/900
